Parquet/src/main/java/io/deephaven/parquet/ParquetFileReader.java

```diff
@@ -77,52 +73,66 @@ public ParquetFileReader(String filePath, SeekableChannelsProvider channelsProvi
         type = fromParquetSchema(fileMetaData.schema,fileMetaData.column_orders);
     }
 
-    private HashSet<String> columnsWithDictionaryUsedOnEveryDataPage = null;
-    public HashSet<String> getColumnsWithDictionaryUsedOnEveryDataPage() {
+    private Set<String> columnsWithDictionaryUsedOnEveryDataPage = null;
+    public Set<String> getColumnsWithDictionaryUsedOnEveryDataPage() {
         if (columnsWithDictionaryUsedOnEveryDataPage == null) {
             columnsWithDictionaryUsedOnEveryDataPage = calculateColumnsWithDictionaryUsedOnEveryDataPage();
         }
         return columnsWithDictionaryUsedOnEveryDataPage;
     }
 
-    private HashSet<String> calculateColumnsWithDictionaryUsedOnEveryDataPage() {
+    /** True only if we are certain every data page in this column chunk uses dictionary encoding;
+     *  note false also covers the "we can't tell" case. */
+    private boolean columnChunkUsesDictionaryOnEveryPage(final ColumnChunk columnChunk) {
+        final ColumnMetaData columnMeta = columnChunk.getMeta_data();
+        if (columnMeta.encoding_stats == null) {
+            return false;  // this is false as "don't know".
+        }
+        for (PageEncodingStats encodingStat : columnMeta.encoding_stats) {
+            if (encodingStat.page_type != PageType.DATA_PAGE
+                    && encodingStat.page_type != PageType.DATA_PAGE_V2) {
+                // skip non-data pages.
+                continue;
+            }
+            // this is a data page.
+            if (encodingStat.encoding != Encoding.PLAIN_DICTIONARY
+                    && encodingStat.encoding != Encoding.RLE_DICTIONARY) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private Set<String> calculateColumnsWithDictionaryUsedOnEveryDataPage() {
         final HashSet<String> result = new HashSet<>(fileMetaData.getSchemaSize());
```

Comment:
`HashSet`->`Set`, but only if you have to edit this file for another reason.

Reply:
Done.